docs(v2): Add step by step in `IncredibleSquaring`

damiramirez opened the pull request:
### What Changed?
This PR adds step by step documentation to the example.

The order of the steps in the crates docs has been updated

### Reviewer Checklist

- [ ] New features are tested and documented
- [ ] PR updates the changelog with a description of changes
- [ ] PR has one of the `changelog-X` labels (if applies)
- [ ] Code deprecates any old functionality before removing it

damiramirez committed:
docs: add docs to the aggregator bin

damiramirez committed:
docs: add docs to the operator bin

damiramirez committed:
docs: add docs to the challenger bin

damiramirez committed:
docs: fix some typos

damiramirez committed:
chore: remove allow

maximopalopoli commented on examples/incredible-squaring/src/bin/task_spammer.rs:
```diff
@@ -9,25 +9,36 @@ use eigensdk::task_spammer::TaskSpammerBuilder;
+        .with_iter((0..).map(U256::from)) // (4) Define an iterator
+        .with_quorum(50, vec![0]) // (5) Set the quorum and threshold percentage
+        .with_interval(Duration::from_secs(10)) // (6) Set the interval to wait between spamming tasks
+        .build() // (7) Build the `TaskSpammer`
```
Maybe we can wrap all these steps in only one

damiramirez replied:
I want to keep consistency with the docs in `lib.rs`, which is why I included the numbers here. Maybe I can only put the numbers like 
```rust
        .with_iter((0..).map(U256::from)) // (4)
        .with_quorum(50, vec![0]) // (5) 
        .with_interval(Duration::from_secs(10)) // (6) 
        .build() // (7)
```

maximopalopoli commented on examples/incredible-squaring/src/lib.rs:
```diff
@@ -15,15 +15,27 @@ use eigensdk::task_manager::TaskManagerError;
 
+/// Implement the [`TaskManagerDefs`] trait for a unit struct.
+/// You need to specify the input and output types of the task. In this case, U256.
+/// You also need to specify the selectors for the new task event and the task responded event.
```
```suggestion
/// You also need to define the selectors for the new task event and the task responded event.
```
To avoid using specify twice.

maximopalopoli reviewed (approved):
LGTM, maybe we can try to avoid repeating the steps at the beginning and during the function, but it's not a real issue.

damiramirez committed:
Update examples/incredible-squaring/src/lib.rs

Co-authored-by: Maximo Palopoli <96491141+maximopalopoli@users.noreply.github.com>

MegaRedHand commented on examples/incredible-squaring/src/bin/challenger.rs:
```diff
@@ -17,26 +15,45 @@ use incredible_squaring::{
 };
+use std::str::FromStr;
 
+/// This example shows how to initialize a challenger and start processing tasks.
```
Can we add this as module documentation (i.e. `//!`)?

MegaRedHand commented on examples/incredible-squaring/src/bin/challenger.rs:
```diff
@@ -17,26 +15,45 @@ use incredible_squaring::{
+/// This example shows how to initialize a challenger and start processing tasks.
+/// Follow the [`Challenger`] documentation to set up a challenger.
+///
+/// 1. Define your types for the task manager (Done in [`ISTaskManager`](incredible_squaring::ISTaskManager))
+/// 2. Create the [`ChallengerConfig`]
+/// 3. Define the task verification logic (Done in [`square`])
+/// 4. Instantiate the task manager instance from your bindings
+/// 5. Build the [`ResponseCalculator`](eigensdk::task_manager::response_calculator::ResponseCalculator)
+/// 6. Create the verifier with [`verifier_from_compute_function`]
+/// 7. Initialize the [`IndexingChallengerProcessor`]
+/// 8. Create and start the challenger
```
Let's keep this only as inlined comments.

MegaRedHand commented on examples/incredible-squaring/src/bin/challenger.rs:
```diff
@@ -17,26 +15,45 @@ use incredible_squaring::{
 #[tokio::main]
 async fn main() {
     init_logger(LogLevel::Info);
+
+    // 2. Create the `ChallengerConfig`
```
Here we can do:

```suggestion
    // 1. Define your types for the task manager (we do this in [`ISTaskManager`](incredible_squaring::ISTaskManager))

    // 2. Create the `ChallengerConfig`
```

damiramirez replied:
I was writing it as doc so I could link to the `ISTaskManager`. I cannot link using commets (`//`)

damiramirez replied:
Maybe something like this?

```
// 1. Define your types for the task manager (we do this in `ISTaskManager`: lib.rs)
```

damiramirez committed:
Merge remote-tracking branch 'origin/v2-dev-2' into docs/improve-is

damiramirez committed:
docs: move to module docs

damiramirez commented:
@MegaRedHand I moved the documentation into the module docs and added inline comments [3039030](https://github.com/Layr-Labs/eigensdk-rs/pull/528/commits/3039030c55c781bffb98e02338e91f53908f6c5d).

damiramirez commented on examples/incredible-squaring/src/bin/aggregator.rs:
```diff
@@ -1,4 +1,5 @@
-#![allow(missing_docs)]
+//! This example shows how to initialize an aggregator
+//! Follow the [`Aggregator`] documentation to set up an aggregator.
```
Would it be better to link to the [module docs](https://github.com/Layr-Labs/eigensdk-rs/blob/v2-dev-2/crates/aggregator/src/lib.rs#L1-L84) instead? The current link points to the `Aggregator` struct.

MegaRedHand replied:
Yes, I think it'd be better

damiramirez replied:
Done [5a0cedc](https://github.com/Layr-Labs/eigensdk-rs/pull/528/commits/5a0cedc04eead768be1bfff5ade256ad6ba071fb)

damiramirez committed:
docs: change link to module docs

MegaRedHand commented on examples/incredible-squaring/src/bin/aggregator.rs:
```diff
@@ -1,5 +1,6 @@
 //! This example shows how to initialize an aggregator
-//! Follow the [`Aggregator`] documentation to set up an aggregator.
+//! Follow the [`Aggregator`](https://github.com/Layr-Labs/eigensdk-rs/blob/v2-dev-2/crates/aggregator/src/lib.rs#L1-L84)
+//! documentation to set up an aggregator.
```
Nitpick

```suggestion
//! Follow the [`eigen-aggregator` crate documentation](https://github.com/Layr-Labs/eigensdk-rs/blob/v2-dev-2/crates/aggregator/src/lib.rs#L1-L84)
//! to set up an aggregator.
```

MegaRedHand commented on examples/incredible-squaring/src/bin/challenger.rs:
```diff
@@ -1,5 +1,6 @@
 //! This example shows how to initialize a challenger and start processing tasks.
-//! Follow the [`Challenger`] documentation to set up a challenger.
+//! Follow the [`Challenger`](https://github.com/Layr-Labs/eigensdk-rs/blob/v2-dev-2/crates/challenger/src/lib.rs#L1-L112)
```
Same

damiramirez replied:
Done [5d80025](https://github.com/Layr-Labs/eigensdk-rs/pull/528/commits/5d80025f9cfe86429743089c7cf37737a9d5a8c4)

damiramirez committed:
Update examples/incredible-squaring/src/bin/aggregator.rs

Co-authored-by: Tomás Grüner <47506558+MegaRedHand@users.noreply.github.com>

damiramirez committed:
docs: change link name
